modules/core/src/main/java/org/apache/ignite/internal/version/ProductVersionSource.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.version;
+
+import org.apache.ignite.internal.properties.IgniteProductVersion;
+
+/**
+ * Allows to obtain product name and version.
+ */
+public interface ProductVersionSource {
```

Comment:
Maybe rename to `IgniteProductVersionSource` ?

Reply:
It's about not just Ignite, it's about any derivative product (like Gridgain or whatnot), so it seems a bit weird to have `Ignite` in the name

Reply:
Yes, I understand, it’s just that when I see such an interface, a question immediately arises in my head: what product are we talking about?

Second point, in the interface there is a method `IgniteProductVersion productVersion();` which obviously returns the ignite product, but inside its name can be "GridGain" which is an extension of ignite. So it seemed reasonable to me to suggest another name for your interface, which in my opinion looks clearer and more obvious. What do you think?

Reply:
Renamed new interface and its implementations